packages/fuselage-ui-kit/src/elements/ChannelsSelectElement/hooks/useChannelsData.ts

```diff
@@ -0,0 +1,46 @@
+import { useEndpoint } from '@rocket.chat/ui-contexts';
+import { useQuery } from '@tanstack/react-query';
+
+type useChannelsDataProps = {
+  filter: string;
+};
+
+type useChannelsDataReturn = {
+  _id: string;
+  t: string;
+  name?: string;
+  fname?: string;
+  avatarETag?: string;
+}[];
+
+const generateQuery = (
+  term = ''
+): {
+  selector: string;
+} => ({ selector: JSON.stringify({ name: term }) });
+
+export const useChannelsData = ({
+  filter,
+}: useChannelsDataProps): useChannelsDataReturn => {
+  const getRooms = useEndpoint(
+    'GET',
+    '/v1/rooms.autocomplete.channelAndPrivate'
+  );
+
+  const { data } = useQuery(
+    ['rooms.autocomplete.channelAndPrivate', filter],
+    () => getRooms(generateQuery(filter)),
+    {
+      keepPreviousData: true,
+    }
+  );
+
+  const result = data?.items.map(({ fname, name, _id, avatarETag, t }) => ({
```

Comment:
This map should be inside the fetch funtion or as `select` argument of the useQuery Hook. Btw, why are you mapping it twice? (once here and once more in the components). Why not already map the information in the structure that the elements will consume?

Reply:
> inside the fetch funtion or as select argument of the useQuery Hook

This still hasn't been implemented. I'll leave this comment here for history purposes